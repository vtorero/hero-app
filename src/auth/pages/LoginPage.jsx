import { useContext } from 'react';
import { useNavigate } from 'react-router-dom';
import { AuthContext } from '../context/AuthContext';

export const LoginPage = () => {

  const {login} = useContext(AuthContext);
  const navigate = useNavigate();

  const onLogin = ()=>{

   const lastPath = localStorage.getItem('lastPath') || '/';
   login('Victor Jimenez');

   navigate(lastPath,{
    replace:true
   })
}

const salir =()=>{
  logout();
}

  return (
    <div className="container">
      <h1>Login</h1>
      <hr />
      <button
      className="btn btn-primary"
      onClick={onLogin}
      >
        Login
      </button>
    </div>
  )
}
