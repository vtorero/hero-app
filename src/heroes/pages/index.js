export * from './MarvelPage';
export * from './DcPages';
export * from './SearchPage';
export * from './HeroPage';