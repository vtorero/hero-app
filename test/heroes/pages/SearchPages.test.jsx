import { fireEvent, render,screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { SearchPage } from '../../../src/heroes/pages/SearchPage';

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom',()=>({
    ...jest.requireActual('react-router-dom'),
    useNavigate:() =>mockedUseNavigate,
}))


beforeEach(()=>jest.clearAllMocks())

describe('Pruebas en el <SearchPage/>', () => {

    test('debe de mostrarse correctamente con los valores por defecto', () => {

        const container = render(
            <MemoryRouter>
                <SearchPage/>
            </MemoryRouter>

        );
        expect(container).toMatchSnapshot();


    })

    test('debe de mostrar a Batman y el input con el valor del querystring', () => {

        const container = render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage/>
            </MemoryRouter>

        );
      //  screen.debug()
        const inputValue = screen.getByRole('textbox');
        expect(inputValue.value).toBe('batman')

        const img = screen.getByRole('img');
        expect(img.src).toContain('/assets/heroes/dc-batman.jpg')

        const alert = screen.getByLabelText('div-search');
        expect(alert.style.display).toBe('none')


        expect(container).toMatchSnapshot();


    })

    test('debe de mostrar un error sino encuentra a batman', () => {

        const container = render(
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <SearchPage/>
            </MemoryRouter>

        );
      //  screen.debug()

        const alert = screen.getByLabelText('div-alert');
        //console.log(alert.style.display).toBe('');
        expect(alert.style.display).toBe('')


        expect(container).toMatchSnapshot();


    })



     test('debe de llamar una pantalla nueva', () => {

       const inputValue='superman';
        render(
            <MemoryRouter initialEntries={['/search']}>
                <SearchPage/>
            </MemoryRouter>);

        const input= screen.getByRole('textbox');
        const form= screen.getByLabelText('form');
        fireEvent.change(input,{target:{name:'searchText',value:inputValue}})
console.log(input.value)
fireEvent.submit(form);
expect(mockedUseNavigate).toHaveBeenCalledWith(`?q=${inputValue}`);

    //console.log(form)


      })

 })