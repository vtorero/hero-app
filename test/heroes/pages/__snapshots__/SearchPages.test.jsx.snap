// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <SearchPage/> debe de mostrar a Batman y el input con el valor del querystring 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        Search Page
      </h1>
      <hr />
      <div
        class="row"
      >
        <div
          class="col-5"
        >
          <h4>
            Searching
          </h4>
          <hr />
          <form
            aria-label="form"
          >
            <input
              autocomplete="off"
              class="form-control"
              name="searchText"
              placeholder="Search a hero"
              type="text"
              value="batman"
            />
            <button
              class="btn btn-outline-primary mt-2"
            >
              Search
            </button>
          </form>
        </div>
        <div
          class="col-7"
        >
          <h4>
            Results
          </h4>
          <hr />
          <div
            aria-label="div-search"
            class="alert alert-primary animate__animated animate__fadeIn"
            style="display: none;"
          >
            Search a hero 
          </div>
          <div
            aria-label="div-alert"
            class="alert alert-danger animate__animated animate__fadeIn"
            style="display: none;"
          >
             No hero with 
            <b>
              batman
            </b>
             
          </div>
          <div
            class="col"
          >
            <div
              class="card animate__animated animate__fadeIn"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-4 "
                >
                  <img
                    alt="Batman"
                    class="card-img"
                    src="assets/heroes/dc-batman.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Batman
                    </h5>
                    <p
                      class="card-text"
                    >
                      Bruce Wayne
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                        Detective Comics #27
                      </small>
                    </p>
                    <a
                      href="/hero/dc-batman"
                    >
                      Mas..
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h1>
      Search Page
    </h1>
    <hr />
    <div
      class="row"
    >
      <div
        class="col-5"
      >
        <h4>
          Searching
        </h4>
        <hr />
        <form
          aria-label="form"
        >
          <input
            autocomplete="off"
            class="form-control"
            name="searchText"
            placeholder="Search a hero"
            type="text"
            value="batman"
          />
          <button
            class="btn btn-outline-primary mt-2"
          >
            Search
          </button>
        </form>
      </div>
      <div
        class="col-7"
      >
        <h4>
          Results
        </h4>
        <hr />
        <div
          aria-label="div-search"
          class="alert alert-primary animate__animated animate__fadeIn"
          style="display: none;"
        >
          Search a hero 
        </div>
        <div
          aria-label="div-alert"
          class="alert alert-danger animate__animated animate__fadeIn"
          style="display: none;"
        >
           No hero with 
          <b>
            batman
          </b>
           
        </div>
        <div
          class="col"
        >
          <div
            class="card animate__animated animate__fadeIn"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4 "
              >
                <img
                  alt="Batman"
                  class="card-img"
                  src="assets/heroes/dc-batman.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Batman
                  </h5>
                  <p
                    class="card-text"
                  >
                    Bruce Wayne
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      Detective Comics #27
                    </small>
                  </p>
                  <a
                    href="/hero/dc-batman"
                  >
                    Mas..
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Pruebas en el <SearchPage/> debe de mostrar un error sino encuentra a batman 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        Search Page
      </h1>
      <hr />
      <div
        class="row"
      >
        <div
          class="col-5"
        >
          <h4>
            Searching
          </h4>
          <hr />
          <form
            aria-label="form"
          >
            <input
              autocomplete="off"
              class="form-control"
              name="searchText"
              placeholder="Search a hero"
              type="text"
              value="batman123"
            />
            <button
              class="btn btn-outline-primary mt-2"
            >
              Search
            </button>
          </form>
        </div>
        <div
          class="col-7"
        >
          <h4>
            Results
          </h4>
          <hr />
          <div
            aria-label="div-search"
            class="alert alert-primary animate__animated animate__fadeIn"
            style="display: none;"
          >
            Search a hero 
          </div>
          <div
            aria-label="div-alert"
            class="alert alert-danger animate__animated animate__fadeIn"
          >
             No hero with 
            <b>
              batman123
            </b>
             
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h1>
      Search Page
    </h1>
    <hr />
    <div
      class="row"
    >
      <div
        class="col-5"
      >
        <h4>
          Searching
        </h4>
        <hr />
        <form
          aria-label="form"
        >
          <input
            autocomplete="off"
            class="form-control"
            name="searchText"
            placeholder="Search a hero"
            type="text"
            value="batman123"
          />
          <button
            class="btn btn-outline-primary mt-2"
          >
            Search
          </button>
        </form>
      </div>
      <div
        class="col-7"
      >
        <h4>
          Results
        </h4>
        <hr />
        <div
          aria-label="div-search"
          class="alert alert-primary animate__animated animate__fadeIn"
          style="display: none;"
        >
          Search a hero 
        </div>
        <div
          aria-label="div-alert"
          class="alert alert-danger animate__animated animate__fadeIn"
        >
           No hero with 
          <b>
            batman123
          </b>
           
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Pruebas en el <SearchPage/> debe de mostrarse correctamente con los valores por defecto 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        Search Page
      </h1>
      <hr />
      <div
        class="row"
      >
        <div
          class="col-5"
        >
          <h4>
            Searching
          </h4>
          <hr />
          <form
            aria-label="form"
          >
            <input
              autocomplete="off"
              class="form-control"
              name="searchText"
              placeholder="Search a hero"
              type="text"
              value=""
            />
            <button
              class="btn btn-outline-primary mt-2"
            >
              Search
            </button>
          </form>
        </div>
        <div
          class="col-7"
        >
          <h4>
            Results
          </h4>
          <hr />
          <div
            aria-label="div-search"
            class="alert alert-primary animate__animated animate__fadeIn"
          >
            Search a hero 
          </div>
          <div
            aria-label="div-alert"
            class="alert alert-danger animate__animated animate__fadeIn"
            style="display: none;"
          >
             No hero with 
            <b />
             
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h1>
      Search Page
    </h1>
    <hr />
    <div
      class="row"
    >
      <div
        class="col-5"
      >
        <h4>
          Searching
        </h4>
        <hr />
        <form
          aria-label="form"
        >
          <input
            autocomplete="off"
            class="form-control"
            name="searchText"
            placeholder="Search a hero"
            type="text"
            value=""
          />
          <button
            class="btn btn-outline-primary mt-2"
          >
            Search
          </button>
        </form>
      </div>
      <div
        class="col-7"
      >
        <h4>
          Results
        </h4>
        <hr />
        <div
          aria-label="div-search"
          class="alert alert-primary animate__animated animate__fadeIn"
        >
          Search a hero 
        </div>
        <div
          aria-label="div-alert"
          class="alert alert-danger animate__animated animate__fadeIn"
          style="display: none;"
        >
           No hero with 
          <b />
           
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
